### PR TITLE
fix(esl-core): register checks improvements

### PR DESCRIPTION
**Pull request by ala-n**

### CODE REVIEW

**Review by julia-murashko — APPROVED** [review]

File: src/modules/esl-base-element/core/esl-base-element.ts [hunk 1]
```diff
@@ -108,16 +108,16 @@ export abstract class ESLBaseElement extends HTMLElement implements ESLBaseCompo
    */
   public static register(this: typeof ESLBaseElement, tagName?: string): void {
     tagName = tagName || this.is;
-    if (!tagName) throw new Error('Can not define custom element');
-    const constructor: any = customElements.get(tagName);
-    if (constructor) {
-      if (constructor.is !== tagName) throw new Error('Element declaration tag inconsistency');
-      return;
+    if (!tagName) {
+      throw new DOMException('[ESL]: Incorrect tag name', 'NotSupportedError');
```
Comment (dshovchko) [inline src/modules/esl-base-element/core/esl-base-element.ts:112]: Maybe a should be implemented message formatting method for consistency purposes?

**Review by dshovchko — APPROVED** [review]

File: src/modules/esl-base-element/test/element.test.ts [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import '../../../polyfills/es5-target-shim';
 import {ESLBaseElement} from '../core';
+import {raw} from 'concurrently/dist/src/defaults';
```
Comment (abarmina) [inline src/modules/esl-base-element/test/element.test.ts:3]: as I see you don't use this import

File: src/modules/esl-base-element/test/element.test.ts [hunk 3]
```diff
@@ -28,17 +29,15 @@ describe('ESLBaseElement', () => {
...
-      .then(() => {
-        expect(() => TestElement2.register('test-test')).not.toThrowError();
-        TestElement2.is = 'test-test-2';
-        // Tag inconsistency
-        expect(() => TestElement2.register('test-test')).toThrowError();
-      });
+    await customElements.whenDefined('test-test');
+    expect(() => TestElement2.register('test-test')).not.toThrowError();
+    expect(() => TestElement2.register('test-test-2')).toThrowError();
+    try {TestElement2.is = 'test-test-2';} catch {}
```
Comment (abarmina) [inline src/modules/esl-base-element/test/element.test.ts:39]: shall we add something to ```catch```?

**Review by abarmina — APPROVED** [review]

File: src/modules/esl-base-element/test/element.test.ts (hunk 2 above)
Comment (ala-n) [reply src/modules/esl-base-element/test/element.test.ts:3]: oh **, missed that
Thanks a lot Ann ❤️

File: src/modules/esl-base-element/test/element.test.ts (hunk 3 above)
Comment (ala-n) [reply src/modules/esl-base-element/test/element.test.ts:39]: I think not.
I'd make an expectation statement here but there is an interesting thing: Node.js throws an error when you try to write immutable property, while browsers just ignore it...

File: src/modules/esl-base-element/core/esl-base-element.ts (hunk 1 above)
Comment (ala-n) [reply src/modules/esl-base-element/core/esl-base-element.ts:112]: Agree I will create a technical cleanup regarding the unification of exception handling

**Pull request merged**